igvm/vm.py

```diff
@@ -378,7 +379,14 @@ def aws_delete(self):
             raise VMError(e)
 
     def is_running(self):
-        return self.hypervisor.vm_running(self)
+        if self.dataset_obj['datacenter_type'] == 'kvm.dct':
+            return self.hypervisor.vm_running(self)
+        elif self.dataset_obj['datacenter_type'] == 'aws.dct':
```

Comment:
no need for an `elif` here. you are already returning above. if you want to make sure the `datacenter_type` is actually supported then you could check it right in the beginning of the function and otherwise raise an error.

Reply:
reworked this function